pkg/pool-weighted/contracts/managed/ManagedPoolSwapFeesLib.sol

```diff
@@ -1,104 +0,0 @@
-        uint256 endSwapFeePercentage
-    );
-
-    uint256 private constant _MIN_SWAP_FEE_PERCENTAGE = 1e12; // 0.0001%
-
-    // The swap fee cannot be 100%: calculations that divide by (1-fee) would revert with division by zero.
-    // Swap fees close to 100% can still cause reverts when performing join/exit swaps, if the calculated fee
-    // amounts exceed the pool's token balances in the Vault. 80% is a very high, but relatively safe maximum value.
-    uint256 private constant _MAX_SWAP_FEE_PERCENTAGE = 80e16; // 80%
-
-    function validateSwapFeePercentage(uint256 swapFeePercentage) internal pure {
-        _require(swapFeePercentage >= _MIN_SWAP_FEE_PERCENTAGE, Errors.MIN_SWAP_FEE_PERCENTAGE);
-        _require(swapFeePercentage <= _MAX_SWAP_FEE_PERCENTAGE, Errors.MAX_SWAP_FEE_PERCENTAGE);
-    }
-
-    /**
-     * @notice Encodes a new swap fee percentage into the provided Pool state.
-     * @dev The swap fee must be within the bounds set by MIN_SWAP_FEE_PERCENTAGE/MAX_SWAP_FEE_PERCENTAGE.
-     * Emits the SwapFeePercentageChanged event.
-     * @return poolState - The modified Pool state with the updated swap fee data. It's the responsiblity of the caller
-     * to write this to storage so this value is persisted.
-     */
-    function setSwapFeePercentage(bytes32 poolState, uint256 swapFeePercentage) internal returns (bytes32) {
-        validateSwapFeePercentage(swapFeePercentage);
-
-        emit SwapFeePercentageChanged(swapFeePercentage);
-
-        return
-            ManagedPoolStorageLib.setSwapFeeData(
-                poolState,
-                block.timestamp,
-                block.timestamp,
-                swapFeePercentage,
-                swapFeePercentage
-            );
-    }
-
-    /**
-     * @notice Encodes a gradual swap fee update into the provided Pool state.
-     * @param startTime - The timestamp when the swap fee change will begin.
-     * @param endTime - The timestamp when the swap fee change will end (must be >= startTime).
-     * @param startSwapFeePercentage - The starting value for the swap fee change.
-     * @param endSwapFeePercentage - The ending value for the swap fee change. If the current timestamp >= endTime,
-     * `getSwapFeePercentage()` will return this value.
-     * @return poolState - The modified Pool state with the updated swap fee data. It's the responsiblity of the caller
-     * to write this to storage so this value is persisted.
-     */
-    function startGradualSwapFeeChange(
-        bytes32 poolState,
-        uint256 startTime,
-        uint256 endTime,
-        uint256 startSwapFeePercentage,
-        uint256 endSwapFeePercentage
-    ) internal returns (bytes32) {
-        validateSwapFeePercentage(startSwapFeePercentage);
-        validateSwapFeePercentage(endSwapFeePercentage);
-
-        if (startSwapFeePercentage != ManagedPoolStorageLib.getSwapFeePercentage(poolState)) {
-            poolState = setSwapFeePercentage(poolState, startSwapFeePercentage);
```

Comment:
Note for the reviewer: with these changes, we are not emitting `SwapFeePercentageChanged` event when the start swap fee is different from the current one. Let's double check if this is alright before merging.

Reply:
This is fine, If you're tracking the swap fee through events then you can just ignore these events entirely and just listen to the gradual update events.